ironfish/src/wallet/account.ts

```diff
@@ -186,29 +186,26 @@ export class Account {
         await this.walletDb.saveDecryptedNote(this, spentNoteHash, spentNote, tx)
       }
 
-      await this.saveConnectedMintsToAssetsStore(transaction, tx)
-      await this.saveConnectedBurnsToAssetsStore(transaction, tx)
+      transactionValue = {
+        transaction,
+        blockHash,
+        sequence,
+        submittedSequence,
+        timestamp,
+        assetBalanceDeltas,
+      }
 
-      await this.walletDb.saveTransaction(
-        this,
-        transaction.hash(),
-        {
-          transaction,
-          blockHash,
-          sequence,
-          submittedSequence,
-          timestamp,
-          assetBalanceDeltas,
-        },
-        tx,
-      )
+      await this.saveConnectedMintsToAssetsStore(transactionValue, tx)
+      await this.saveConnectedBurnsToAssetsStore(transactionValue, tx)
+
+      await this.walletDb.saveTransaction(this, transaction.hash(), transactionValue, tx)
     })
 
     return assetBalanceDeltas
   }
 
   async saveConnectedMintsToAssetsStore(
-    transaction: Transaction,
+    { blockHash, sequence, transaction }: TransactionValue,
```

Comment:
I didn't know you could do this in TypeScript 🤯

Reply:
:cowboy_hat_face: